### PR TITLE
Allow array of full ip range generator to support other netmasks than /24

### PR DESCRIPTION
**Pull request by dkranke**
The array of full ip range generator was limited to /24 netmasks like 192.168.1.0/24.
Using the netmask package it now supports all kinds of netmasks like 10.0.0.0/12 or 172.16.1/21.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -40,16 +42,24 @@ function Arpping({ timeout = 5, includeEndpoints = false, useCache = true, cache
...
     // don't use default assignment so false-y values are overwritten
     ip = ip || this.myIP;
-    var ipStart = ip.substr(0, ip.lastIndexOf('.') + 1);
+    // add mask to ip eg xxx.xxx.xxx.xxx/yy
+    ip = ip + '/' + mask;
+
+    var range = [];
+    var block = new Netmask(ips);
+    range.push(block.base);
+    range.forEach((ip, _, index) => range.push(ip));
```
Comment (Tubbie) [inline index.js]: @dkranke Shouldn't this be block.forEach?

File: index.js [hunk 2]
```diff
@@ -40,16 +42,24 @@ function Arpping({ timeout = 5, includeEndpoints = false, useCache = true, cache
...
-Arpping.prototype._getFullRange = function(ip) {
+Arpping.prototype._getFullRange = function(ip, mask = 24) {
     // don't use default assignment so false-y values are overwritten
     ip = ip || this.myIP;
-    var ipStart = ip.substr(0, ip.lastIndexOf('.') + 1);
+    // add mask to ip eg xxx.xxx.xxx.xxx/yy
+    ip = ip + '/' + mask;
+
+    var range = [];
+    var block = new Netmask(ips);
```
Comment (Tubbie) [inline index.js]: @dkranke This should be ip instead of ips.

File: index.js (hunk 1 above)
Comment (dkranke) [reply index.js]: Your right, i actually only wrote the code and didn't try it.
I'm still in the making of a project that needs it.

File: index.js (hunk 2 above)
Comment (dkranke) [reply index.js]: Will fix both issues asap.

File: index.js (hunk 1 above)
Comment (Tubbie) [reply index.js]: Thank you for your quick reply. 
I found out these issues while I was testing your PR.  :)

Comment (dkranke) [reply index.js]: I've learned that I should always test before creating a PR

**Comment by haf-decent** [comment]
Hey, thanks for pointing this out and taking the time to implement it. I'm not really sure what best practice is here, but I noticed some little things in your PR that needed to be changed that I didn't want to make you do so I created a new branch, pr/8, and added your PR + those changes. 

The changes included things like:
- passing the `mask` to the discover function if it retries after running findMyInfo (index: line 104 or so)
- parsing the netmask from the ipconfig/ifconfig command (you had "...indexOf('netmask ') **+ 5**...", when it needs to be "**...+ 8...**" to account for the full search string, plus on my machine the netmask returned as hex so I had to add an extra parser function to convert that to a usable string when needed)
- passing the `ip` and `mask` separately to Netmask rather than building a string (more my own preference than anything)
- merging @Tubbie 's previous PR.

Again, not sure if I'm supposed to add a PR to your PR with my updates to then merge in or if I should just merge the new branch in on its own and close this PR. Let me know if you have a preference for any reason. Oh, also please test that new branch to make sure it works properly on your machine before I merge anything.

**Comment by dkranke** [comment]
My implementation was a very dirty hack and I didn't even had the time to test it and create a example which I really should have been doing before creating this pull request. So you and Tubbie pointed out some issues. For me it's the first time working on a public project and create a pull request, but I also have no idea. Also thanks to you for creating the base package.

**Comment by haf-decent** [comment]
I believe this issue is now addressed with #12 
Thanks for the help @dkranke and @Tubbie!

**Pull request closed without merging**